bascule/key/cache.go

```diff
@@ -137,14 +152,24 @@ func (cache *multiCache) copyPairs() map[string]Pair {
 	return newPairs
 }
 
-func (cache *multiCache) ResolveKey(keyID string) (pair Pair, err error) {
+func (cache *multiCache) ResolveKey(ctx context.Context, keyID string) (pair Pair, err error) {
 	var ok bool
 	pair, ok = cache.fetchPair(keyID)
 	if !ok {
 		cache.update(func() {
 			pair, ok = cache.fetchPair(keyID)
+			if isContextDone(ctx) {
+				err = ErrOperationTimedOut
+				return
+			}
+
 			if !ok {
-				pair, err = cache.delegate.ResolveKey(keyID)
+				pair, err = cache.delegate.ResolveKey(ctx, keyID)
+				if isContextDone(ctx) {
+					err = ErrOperationTimedOut
```

Comment:
When a context gets canceled, you should always return `ctx.Err()`.  That's the general contract of methods that accept a `context` object.  It lets a client unambiguously determine if it was a contextual cancelation by comparing the error you return against one of the public errors in the `context` package.